### PR TITLE
build: spinner and api's

### PR DESCRIPTION
**Pull request by shirishBajpai**

### CODE REVIEW

File: src/Components/Product/ProductList/ProductList.js [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
-        .then(response => {
-            console.log(response);
-        });
-const ProductList = () => {
-    
-    return (
-        <div>
+function ProductList() {
+    const [error, setError] = useState(null);
+    const [isLoaded, setIsLoaded] = useState(false);
```
Comment (abhinavRai23) [inline src/Components/Product/ProductList/ProductList.js:7]: The initial state could be `true`. So by default component will start with spinner until the API call gets resolved.

File: src/Components/Product/ProductList/ProductList.js [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+    } else {
+        return (
+            <div>
+                {items.map((user) => (
+                    <div className="container">
+                        <Row className="productList">
+                            <div key={user.id}></div>
+                            <Col><img src="phone1.png" width="150" height="225" /></Col>
+                            <Col><p className="ProductList_heading">{user.name}</p>
+                                <p className="ProductList_details">{user.brand}<br />{user.RAM}<br />{user.ROM}<br />{user.Expandable}</p>
```
Comment (abhinavRai23) [inline src/Components/Product/ProductList/ProductList.js:41]: could have destructured user objects and keep variable name in camelCase format.

File: src/Components/Product/ProductList/ProductList.js [hunk 3]
```diff
@@ -1,19 +1,52 @@
...
+            .then(
+                (result) => {
+                    setIsLoaded(true);
+                    console.log(result.employee);
+                    setItems(result.employee);
+                },
+                (error) => {
+                    setIsLoaded(true);
+                    setError(error);
+                }
```
Comment (gaurav702) [inline src/Components/Product/ProductList/ProductList.js:22]: chain a .catch to catch the error

File: src/Components/Product/ProductList/ProductList.js [hunk 4]
```diff
@@ -1,19 +1,52 @@
...
+            )
+    }, [])
+
+    if (error) {
+        return <div>Error: {error.message}</div>;
```
Comment (gaurav702) [inline src/Components/Product/ProductList/ProductList.js:27]: add a generic error message like "Something went wrong"

**Pull request merged**